src/vm/method.hpp

```diff
@@ -500,8 +500,10 @@ class MethodDesc
     // Return a pointer to the method dictionary for an instantiated generic method
     // The initial slots in a method dictionary are the type arguments themselves
     // Return NULL if not an instantiated method
-    Dictionary* GetMethodDictionary();
-    DictionaryLayout* GetDictionaryLayout();
+    // These APIs are not multi-threaded safe: the dictionary and dictionary layout pointers
+    // can be updated by other threads during a dictionary size expansion.
+    Dictionary* GetMethodDictionary_Unsafe();
+    DictionaryLayout* GetDictionaryLayout_Unsafe();
```

Comment:
I am not sure whether whether the `_Unsafe` suffix is useful here.

The old pointer to the dictionary layout will still remain valid when there is a race. This is about as unsafe as many other methods on MethodTable and friends. For example, when you ask for a method entrypoint, the method can have a different entrypoint by the time you get it - because of it was JITed in the meantime.

Reply:
Ok, in that case, i'll revert this renaming